packages/api/cms-api/src/builds/builds.resolver.ts

```diff
@@ -1,23 +1,27 @@
 import { V1CronJob } from "@kubernetes/client-node";
+import { Inject } from "@nestjs/common";
 import { Args, Mutation, Query, Resolver } from "@nestjs/graphql";
 
 import { CurrentUserInterface } from "../auth/current-user/current-user";
 import { GetCurrentUser } from "../auth/decorators/get-current-user.decorator";
-import { ContentScopeService } from "../content-scope/content-scope.service";
 import { INSTANCE_LABEL } from "../kubernetes/kubernetes.constants";
 import { KubernetesService } from "../kubernetes/kubernetes.service";
+import { AccessControlService } from "../user-permissions/access-control.service";
+import { RequiredPermission } from "../user-permissions/decorators/required-permission.decorator";
+import { ACCESS_CONTROL_SERVICE } from "../user-permissions/user-permissions.types";
 import { BuildsService } from "./builds.service";
 import { AutoBuildStatus } from "./dto/auto-build-status.object";
 import { Build } from "./dto/build.object";
 import { CreateBuildsInput } from "./dto/create-builds.input";
 import { SkipBuild } from "./skip-build.decorator";
 
 @Resolver(() => Build)
+@RequiredPermission(["pageTree"], { skipScopeCheck: true })
 export class BuildsResolver {
     constructor(
         private readonly kubernetesService: KubernetesService,
         private readonly buildsService: BuildsService,
-        private readonly contentScopeService: ContentScopeService,
+        @Inject(ACCESS_CONTROL_SERVICE) private readonly accessControlService: AccessControlService,
```

Comment:
Do you really need a custom token for injection?

Reply:
Yes, because it is possible to inject a custom `accessControlService` in `forRootAsync`. See https://github.com/vivid-planet/comet/pull/1223/files#diff-87c5a6413e447c9c46e24a82309ca67e8200fd20332af148f99c70344145a287R43